lib/ruby_lsp/ruby_lsp_rspec/rspec_formatter.rb

```diff
@@ -62,6 +62,9 @@ def example_pending(notification)
 
       def stop(notification)
         RubyLsp::LspReporter.instance.shutdown
+        if RubyLsp::LspReporter.start_coverage?
+          RubyLsp::LspReporter.instance.at_coverage_exit
+        end
```

Comment:
While `shutdown` doesn't call `internal_shutdown` when coverage is enabled and we avoided double shutdown that way, I think it's very implicit. It'd be better if we have explicit branching:

```suggestion
        if RubyLsp::LspReporter.start_coverage?
          RubyLsp::LspReporter.instance.at_coverage_exit
        else
          RubyLsp::LspReporter.instance.shutdown
        end
```